### PR TITLE
Add initial implementation for efficientnet-v2-s model

### PR DESCRIPTION
**Pull request by sukyoungjeong-furiosa**
## Changes

Deploy an initial model implementation for efficienent-v2-s, source from [pytorch efficientnet_v2_s](https://pytorch.org/vision/main/models/generated/torchvision.models.efficientnet_v2_s.html).

- accuracy: 75.208% (original f32 model: 84.228%)

```python
import torch
import onnx
from tempfile import NamedTemporaryFile
import furiosa.optimizer.frontend.onnx

torch_model = torch.hub.load(
    "pytorch/vision:v0.14.0", "efficientnet_v2_s", weights="DEFAULT", skip_validation=True, force_reload=True
)
torch_model = torch_model.eval()
with NamedTemporaryFile(suffix=".onnx") as file:
    torch.onnx.export(
        torch_model, torch.randn(1, 3, 384, 384), file, opset_version=13
    )
    onnx_model = onnx.load_model(file.name)

onnx.shape_inference.infer_shapes(onnx_model, check_type=True, strict_mode=True)
onnx.checker.check_model(onnx_model)

optimized_onnx_model = furiosa.optimizer.frontend.onnx.optimize_model(onnx_model)
onnx.checker.check_model(optimized_onnx_model, full_check=True)
optimized_onnx_model = optimized_onnx_model.SerializeToString()
```

### CODE REVIEW

**Comment by furiosa-infra** [comment]
874c39c Add example, unit test, regression test
```

----------------------------------------------------- benchmark: 1 tests ----------------------------------------------------
Name (time in ms)                     Min       Max     Mean  StdDev   Median     IQR   Outliers      OPS  Rounds  Iterations
-----------------------------------------------------------------------------------------------------------------------------
test_efficientnetv2s_accuracy     21.3837  337.9542  34.2150  7.9874  33.6569  3.2938  1508;2318  29.2270   50000           1
-----------------------------------------------------------------------------------------------------------------------------

Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
```

**Comment by hyunsik** [comment]
@sukyoungjeong-furiosa 리뷰를 보면 되는 상황인거죠? 그럼 제가 보겠습니다.

File: furiosa/models/vision/efficientnet_v2_s/__init__.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        image = resize(image, INPUT_SIZE, Image.Resampling.BILINEAR)
+        image = center_crop(image, INPUT_SIZE)
+
+        image = np.ascontiguousarray(image)
+        image = np.transpose(image, (2, 0, 1))
+
+        image = image.astype(np.float32) / 255
+
+        data = normalize(image)
+        return data[np.newaxis, ...], None
```
Comment (furiosamg) [inline furiosa/models/vision/efficientnet_v2_s/__init__.py:70]:
```suggestion
        return np.expand_dims(data, axis=0), None
```

취향 차이이긴 하지만 이런일을 하는 함수가 있습니다. [레퍼런스](https://numpy.org/doc/stable/reference/generated/numpy.expand_dims.html)
그런데 다른 모델들이 이미 이렇게 작성되어 있어서요, 편하실대로 해주시면 될 것 같습니다.

File: furiosa/models/vision/efficientnet_v2_s/__init__.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
+from pathlib import Path
+from typing import Any, Dict, List, Sequence, Tuple
+
+from PIL import Image, ImageOps
+import numpy as np
+import numpy.typing as npt
+
+from furiosa.registry.model import Format, Metadata, Publication
+
+from ...types import ImageClassificationModel, PostProcessor, PreProcessor
```
Comment (furiosamg) [inline furiosa/models/vision/efficientnet_v2_s/__init__.py:10]:
```suggestion
from ...types import ImageClassificationModel, Platform, PostProcessor, PreProcessor
```

밑의 리뷰 커멘트에서 추가로 필요한 의존성입니다.

File: furiosa/models/vision/efficientnet_v2_s/__init__.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
+from pathlib import Path
+from typing import Any, Dict, List, Sequence, Tuple
```
Comment (furiosamg) [inline furiosa/models/vision/efficientnet_v2_s/__init__.py:2]:
```suggestion
from typing import Any, Dict, List, Sequence, Tuple, Type
```

밑의 리뷰 커멘트에서 추가로 필요한 의존성입니다.

File: furiosa/models/vision/efficientnet_v2_s/__init__.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+
+class EfficientNetV2sPostProcessor(PostProcessor):
+    def __call__(self, model_outputs: Sequence[npt.ArrayLike], contexts: Any = None) -> str:
+        return CLASSES[int(np.argsort(model_outputs[0], axis=1)[:, ::-1][0, 0])]
+
+
+class EfficientNetV2s(ImageClassificationModel):
+    """EfficientNetV2-s model"""
+
```
Comment (furiosamg) [inline furiosa/models/vision/efficientnet_v2_s/__init__.py:80]:
```suggestion
    postprocessor_map: Dict[Platform, Type[PostProcessor]] = {
        Platform.PYTHON: EfficientNetV2sPostProcessor,
    }
```

cli 에서 모델 리스트를 출력할 때, postprocessor 매핑 정보도 필요해서요, 이 코드가 필요할 것 같습니다.
이와 별개로 지금 다른 모델에도 빠져있는 것 같은데 이건 별도 이슈에서 제가 진행하겠습니다.

**Review by furiosamg — COMMENTED** [review]
I have one major comment, but in overall I think it's good. Thanks for your contribution

File: furiosa/models/vision/efficientnet_v2_s/__init__.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            enf=artifacts[EXT_ENF],
+            format=Format.ONNX,
+            family="EfficientNetV2",
+            version="s",
+            metadata=Metadata(
+                description="EfficientNetV2s ImageNet-1K",
+                publication=Publication(url="https://arxiv.org/abs/2104.00298"),
+            ),
+            preprocessor=EfficientNetV2sPreProcessor(),
+            postprocessor=EfficientNetV2sPostProcessor(),
```
Comment (furiosamg) [inline furiosa/models/vision/efficientnet_v2_s/__init__.py:100]: 이 부분도 postprocessor_map 에서 가져오도록 해야할 것 같네요

**Comment by furiosa-infra** [comment]
78c403b Address review comments
```

----------------------------------------------------- benchmark: 1 tests ----------------------------------------------------
Name (time in ms)                     Min       Max     Mean  StdDev   Median     IQR   Outliers      OPS  Rounds  Iterations
-----------------------------------------------------------------------------------------------------------------------------
test_efficientnetv2s_accuracy     13.1713  231.6936  18.1714  4.6126  17.7733  1.3691  1067;2972  55.0314   50000           1
-----------------------------------------------------------------------------------------------------------------------------

Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
```

**Comment by sukyoungjeong-furiosa** [comment]
@furiosamg 님, 제안주신 부분들 반영했습니다.

**Review by furiosamg — APPROVED** [review]
LGTM! Thanks
